notebooks/109-performance-tricks/109-latency-tricks.ipynb

```diff
@@ -0,0 +1,460 @@
+    "\n",
+    "if not onnx_path.exists():\n",
+    "    dummy_input = torch.randn(1, 3, IMAGE_HEIGHT, IMAGE_WIDTH)\n",
+    "    torch.onnx.export(pytorch_model, dummy_input, onnx_path)\n",
+    "\n",
+    "onnx_model = core.read_model(onnx_path)\n",
+    "onnx_model = core.compile_model(onnx_model, device_name=\"CPU\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "show_result(model=onnx_model, result=result)\n",
+    "benchmark_model(model=onnx_model, input=input_image, model_name=\"ONNX model\")\n",
+    "\n",
+    "del onnx_model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "### OpenVINO IR model\n",
+    "\n",
+    "Let's convert the ONNX model to OpenVINO Intermediate Representation (IR) and run it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from openvino.tools import mo\n",
+    "\n",
+    "ov_model_fp32 = mo.convert_model(onnx_path)\n",
+    "ov_cpu_model_fp32 = core.compile_model(ov_model_fp32, device_name=\"CPU\")\n",
+    "\n",
+    "show_result(model=ov_cpu_model_fp32, result=result)\n",
+    "benchmark_model(model=ov_cpu_model_fp32, input=input_image, model_name=\"OpenVINO model\")\n",
+    "\n",
+    "del ov_cpu_model_fp32  # release resources"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "### OpenVINO IR FP16 model\n",
+    "\n",
+    "Reducing the precision is one of the well-know methods for faster inference. We could use quantization but in that case we should expect a little accuracy drop. That's why we skip that step in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ov_model = mo.convert_model(onnx_path, compress_to_fp16=True)\n",
+    "ov_cpu_model = core.compile_model(ov_model, device_name=\"CPU\")\n",
+    "\n",
+    "show_result(model=ov_cpu_model, result=result)\n",
+    "benchmark_model(model=ov_cpu_model, input=input_image, model_name=\"OpenVINO FP16 model\")\n",
+    "\n",
+    "del ov_cpu_model  # release resources"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "### OpenVINO IR FP16 model + additional config\n",
+    "\n",
+    "There is a possibility to add a config for any device (CPU in this case). We're going to increase the number of threads to equal number of our cores"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_cores = os.cpu_count()\n",
+    "\n",
+    "ov_cpu_config_model = core.compile_model(ov_model, device_name=\"CPU\", config={\"INFERENCE_NUM_THREADS\": num_cores})\n",
```

Comment:
Hi @adrianboguszewski since sometimes our customers may wish to free the some CPU cores for other non-inference task, this notebook is a real good example to show how to set the configuration for a device. Could you help to show more example about how to set more parameters, for example, the number of stream ? Thanks

Reply:
I tried to use a different number of streams, but it didn't give me a better performance. Have you tried it? Did you get lower latency?

Reply:
Yes, It will not improve the performance. I just want to have a example to show how the set these kind of parameter, cuz many developers are asking about it, and we have no page to explain it except the API document.

Reply:
I see. I can mention other parameters also, but this notebook should be focused on performance tricks as users are not aware of many of them.

Reply:
Understand, changing the number of streams can help to balance the latency and throughout manually. Sometimes we cant just use performance hint to get a best latency or throughout.